### DISCUSSION TITLE
jq parsing of url on get

### DISCUSSION THREAD

**Discussion by rgraham7503** [hyperjumptech/monika#937] [General]
Hi just a question. Is there a way to use JQ inline with the GET request to pre-manipulate the response body? Example...I have a pretty complex one-liner JQ with arguments etc. works fine in shell on RHEL and MacOS...I was thinking, If I could use this in Monika and alert if the body contains "whatever" that would be way better than a shell script. 
I tried it and it doesnt work...but curious if there is a way. I went through all my monika yamls and I dont see I have tried this before.

**Comment by sapiderman** [comment]
Hi @rgraham7503 

JQ here you mean the JSON Processor (https://stedolan.github.io/jq/)? At this time monika doesn't yet support piping i'm afraid. Or the jq plugin as an alert preprocessor. We're always open to any good PRs with interesting use cases however.

**Reply by rgraham7503** [reply]
Hi @sapiderman 

So here is the use-case. There is an api I need to call that will retrieve certificate data and the response is in json. Its array contains the subject name and date of expiry, among other items, but these are the only two I need. The issue is the date is in a format of "19 Nov 2018 17:40:23 GMT" So I had to convert that to epoch, and I need to check for expiring in the next 30 days, so had to implement that logic, then parse the two json fields loop through and if expiring or expired only show those on output. 
Its a huge one liner that works using curl and jq on linux and mac without making a single change. I also have automated this using github actions/workflows. 
What I wanted to do was to add this beautiful one-liner to my Monika monitoring and if response body contains Expire then alert. However I think its to complex of a use-case to be implemented in Monika without the JQ and Date logic.

**Reply by sapiderman** [reply]
It seems you need to monitor the certificate validity of some machine? Would this [TLS checker](https://monika.hyperjump.tech/guides/tls-checkers) work for you.

A possible solution perhaps if your curl-jq pipeline outputs itself into a monika YAML config file somehow. Monika continuously monitors its config file and updates on any changes.

**Reply by rgraham7503** [reply]
Its a SaaS product, that does not have alerting when certs expire. TLS monitor wouldnt work. Its ok, just curious and always looking to integrate with Monika!
